Dockerfile-proxy

```diff
@@ -1,19 +1,21 @@
 ARG RUNTIME_IMAGE=debian:buster-20200514-slim
+ARG BUILDPLATFORM=linux/amd64
 
 # Precompile key slow-to-build dependencies
-FROM golang:1.14.2-alpine as go-deps
+FROM --platform=$BUILDPLATFORM golang:1.14.2-alpine as go-deps
 WORKDIR /linkerd-build
 COPY go.mod go.sum ./
 COPY bin/install-deps bin/
 RUN go mod download
 RUN ./bin/install-deps
```

Comment:
This stage will be ran separately for each arch and `go install` will use that arch, so this is fine. What do you think though about using `ARG TARGETARCH` and passing that to `install-deps`, so it's clearer what's going on, like you did in the stages below?

Reply:
It is possible, but then it will result in 3 different `go-deps` layers. It will get reused on their arch only, not across the arch. That is my understanding. LMKWYT

Reply:
Isn't that stage already producing 3 different layers? This is an extract of the output I see currently:
```
=> CACHED [linux/amd64 go-deps 2/6] WORKDIR /linkerd-build                                                                                                                                                                               0.0s
 => CACHED [linux/amd64 go-deps 3/6] COPY go.mod go.sum ./                                                                                                                                                                                0.0s
 => CACHED [linux/amd64 go-deps 4/6] COPY bin/install-deps bin/                                                                                                                                                                           0.0s
 => CACHED [linux/amd64 go-deps 5/6] RUN go mod download                                                                                                                                                                                  0.0s 
 => CACHED [linux/amd64 go-deps 6/6] RUN ./bin/install-deps                                                                                                                                                                               0.0s 
 => CACHED [linux/amd64 golang 1/7] WORKDIR /linkerd-build                                                                                                                                                                                0.0s 
 => CACHED [linux/amd64 golang 2/7] COPY pkg/flags pkg/flags                                                                                                                                                                              0.0s 
 => CACHED [linux/amd64 golang 3/7] COPY pkg/tls pkg/tls                                                                                                                                                                                  0.0s 
 => CACHED [linux/amd64 golang 4/7] COPY pkg/version pkg/version                                                                                                                                                                          0.0s 
 => CACHED [linux/amd64 go-deps 2/6] WORKDIR /linkerd-build                                                                                                                                                                               0.0s 
 => CACHED [linux/amd64 go-deps 3/6] COPY go.mod go.sum ./                                                                                                                                                                                0.0s
 => CACHED [linux/amd64 go-deps 4/6] COPY bin/install-deps bin/                                                                                                                                                                           0.0s 
 => CACHED [linux/amd64 go-deps 5/6] RUN go mod download                                                                                                                                                                                  0.0s
 => CACHED [linux/amd64 go-deps 6/6] RUN ./bin/install-deps                                                                                                                                                                               0.0s
 => CACHED [linux/amd64 golang 1/7] WORKDIR /linkerd-build                                                                                                                                                                                0.0s
 => CACHED [linux/amd64 golang 2/7] COPY pkg/flags pkg/flags                                                                                                                                                                              0.0s
 => CACHED [linux/amd64 golang 3/7] COPY pkg/tls pkg/tls                                                                                                                                                                                  0.0s
 => CACHED [linux/amd64 golang 4/7] COPY pkg/version pkg/version                                                                                                                                                                          0.0s
 => [linux/amd64 golang 5/7] RUN CGO_ENABLED=0 GOOS=linux GOARCH=arm go build -mod=readonly ./pkg/...                                                                                                                                     5.0s
 => [linux/amd64 golang 5/7] RUN CGO_ENABLED=0 GOOS=linux GOARCH=arm64 go build -mod=readonly ./pkg/...                                                                                                                                   4.9s
 => CACHED [linux/amd64 go-deps 2/6] WORKDIR /linkerd-build                                                                                                                                                                               0.0s
 => CACHED [linux/amd64 go-deps 3/6] COPY go.mod go.sum ./                                                                                                                                                                                0.0s
 => CACHED [linux/amd64 go-deps 4/6] COPY bin/install-deps bin/                                                                                                                                                                           0.0s
 => CACHED [linux/amd64 go-deps 5/6] RUN go mod download                                                                                                                                                                                  0.0s
 => CACHED [linux/amd64 go-deps 6/6] RUN ./bin/install-deps
```
`go-deps` is running 3 times, with the same base `linux/amd64`, but I think each run is for each different arch. Which is fine since the ensuing `golang` stage requires a `go` cache built for each one of the archs.